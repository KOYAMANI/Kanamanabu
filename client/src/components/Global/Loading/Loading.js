import React from 'react'
import { 
    LoadingContainer, 
    LoadingSpinner 
} from './LoadingElenments';


const Loading = ({size = 100}) => {
    return (
        <LoadingContainer>
            <LoadingSpinner 
            width = {size}
            height = {size}
            animation = "border"
            />       
        </LoadingContainer>
    )
}

export default Loading
