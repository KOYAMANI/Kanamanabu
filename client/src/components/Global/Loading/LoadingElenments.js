
import styled from 'styled-components';
import { Spinner } from 'react-bootstrap'

export const LoadingContainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 100%;
`

export const LoadingSpinner = styled(Spinner)`
    width: ${({ width }) => width || "100"};,
    height: ${({ height }) => height || "100"};,
`
