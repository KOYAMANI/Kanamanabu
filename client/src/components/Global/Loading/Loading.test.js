import {cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Loading from "./Loading";

afterEach(cleanup);

describe('Achievement badge', () =>{

    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<Loading/>, div)
    });

    test('match snapshot',()=>{
        const tree = renderer.create(<Loading/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})