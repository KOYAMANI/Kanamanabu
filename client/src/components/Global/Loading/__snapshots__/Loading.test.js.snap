// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievement badge match snapshot 1`] = `
<div
  className="sc-bczRLJ dRMDrP"
>
  <div
    className="sc-gsnTZi kVQEub spinner-border"
    height={100}
    width={100}
  />
</div>
`;
